docs/writing-intel-modules.md

```diff
@@ -0,0 +1,248 @@
+[set to the time that the CLI is run](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/sync.py#L131-L134).
+All `cartography` intel modules need to set the `lastupdated` property on all nodes and all relationships to this
+`update_tag`.  You can see a couple examples of this in our
+[AWS ingestion code](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/intel/aws/__init__.py#L106) and our
+    [GCP ingestion code](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/intel/gcp/__init__.py#L134).
+
+
+#### All nodes need these fields
+
+- <a name="id">`id`</a> - an ID should be a string that uniquely identifies the node.  In AWS, this is usually an
+    ARN.  In GCP, this is usually a partial URI.
+
+    If possible, we should use API-provided fields for IDs and not create our own.
+    In some cases though this is unavoidable -
+    see [GCPNetworkTag](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/docs/schema/gcp.md#gcpnetworktag).
+
+    When setting an `id`, ensure that you also include the field name that it came from.  For example, since we've
+    decided to use `partial_uri`s as an id for a GCPVpc,  we should include both `partial_uri` _and_ `id` on the node.
+    This way, a user can tell what fields were used to derive the `id`.  This is accomplished [here](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/intel/gcp/compute.py#L455-L457).
+
+- `lastupdated` - See [below](#lastupdated-and-firstseen) on how to set this.
+- `firstseen` - See [below](#lastupdated-and-firstseen) on how to set this.
+
+#### All relationships need these fields
+
+Cartography currently does not create indexes on relationships, so we should keep relationships lightweight with only these two fields:
+
+- `lastupdated` - See [below](#lastupdated-and-firstseen) on how to set this.
+- `firstseen` - See [below](#lastupdated-and-firstseen) on how to set this.
+
+
+
+
+#### Run queries only on indexed fields for best performance
+
+In this example of ingesting GCP VPCs, we connect VPCs with GCPProjects
+[based on GCPProject `id`s and GCPVpc `id`s](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/intel/gcp/compute.py#L451).
+`id`s are indexed, as seen [here](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/data/indexes.cypher#L45)
+and [here](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/data/indexes.cypher#L42).
+All of these queries use indexes for faster lookup.
+
+
+#### Create an index for new nodes
+
+Be sure to [update the indexes.cypher file](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/data/indexes.cypher)
```

Comment:
Indexing on ID is required. Indexing on anything else that will be frequently queried is encouraged.